### ISSUE TITLE
Aws-sdk TransferManager produce 412 Error on uploaded multipart files 

### ISSUE THREAD

**Issue by Ebalaitung** [minio/minio#9821]
We have upgraded Minio from 'RELEASE.2019-02-26T19-51-46Z' to 'RELEASE.2020-05-16T01-33-21Z'

In our application we have a piece of code which uploads folder with jar files to minio via aws-sdk Transfer manager like this:
`transferManager.uploadDirectory(bucket, path, file, true)`

Then we download same folder via:
`MultipleFileDownload download = transferManager.downloadDirectory(bucket, path, file, true);
            download.waitForCompletion();`

Before upgrade everything was working fine. But after upgrade we got:

`com.amazonaws.services.s3.model.AmazonS3Exception: Precondition Failed (Service: Amazon S3; Status Code: 412; Error Code: 412 Precondition Failed; Request ID: 161775A73AE10953; S3 Extended Request ID: null), S3 Extended Request ID: null`

We also traced minio and got following error there :

```
 [REQUEST s3.HeadObject] 11:46:46.918
  HEAD /vds-models/folder1/app1.jar?partNumber=1
  Host:  
  X-Forwarded-Proto: https
  Amz-Sdk-Retry: 0/0/500
  Authorization: AWS JOZXPHL1ZC8WXZAJXNU7:5mJ4id0QrXw0wKus/OPbCfcR/X4=
  Connection: close
  Content-Length: 0
  Content-Type: application/octet-stream
  Amz-Sdk-Invocation-Id: abe90f87-a295-c788-1d8a-58236fe1296a
  Date: Thu, 11 Jun 2020 11:46:52 GMT
  User-Agent: aws-sdk-java/1.11.415 Windows_10/10.0 Java_HotSpot(TM)_64-Bit_Server_VM/25.231-b11 java/1.8.0_231
  X-Real-Ip: 10.214.74.83
 
  [RESPONSE] [11:46:46.918] [ Duration 355µs  ↑ 138 B  ↓ 862 B ]
  412 Precondition Failed
  Content-Security-Policy: block-all-mixed-content
  Content-Type: application/xml
  ETag: "5e79b87d8ac36b9e84c2ed935f67418d-23"
  Last-Modified: Wed, 10 Jun 2020 06:05:50 GMT
  Server: MinIO/RELEASE.2020-05-16T01-33-21Z
  X-Amz-Request-Id: 16177ACB4ED2DC1B
  Accept-Ranges: bytes
  Vary: Origin
  X-Xss-Protection: 1; mode=block
  Content-Length: 506
  <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>PreconditionFailed</Code>
		<Message>At least one of the pre-conditions you specified did not hold</Message>
		<Key>folder1/app1.jar</Key>
		<BucketName>vds-models</BucketName>
		<Resource>/vds-models/folder1/app1.jar</Resource>
		<RequestId>16177ACB4ED2DC1B</RequestId>
		<HostId>88fb5a13-6471-47eb-b32e-27ee464684cd</HostId>
		</Error>
```

This issue is only valid for uploaded files via aws-sdk. If we put filed manually to minio file system, files could be downloaded without errors.
Seems like there is an issue with multipart, as uploaded files has corresponding fs.json files in minio.sys folder.

**Comment by harshavardhana** [comment]
This is fixed in latest release PTAL

**Comment by Ebalaitung** [comment]
> This is fixed in latest release PTAL

Could you provide corresponding ticket number?

**Comment by harshavardhana** [comment]
```
commit 2c372a98947365af7a8a2b4bd1c3c0b4d90b9a9e
Author: kannappanr <30541348+kannappanr@users.noreply.github.com>
Date:   Wed Jun 10 09:22:15 2020 -0700

    Send Partscount only when partnumber is specified (#9793)
    
    Fixes #9789
```

**Comment by harshavardhana** [comment]
Closing this as already fixed. please let us know if you see the issue again.
